cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -199,8 +200,11 @@ compute_submap_indices(const IndexMap& imap,
   // their owner in the submap. This is required since not all indices in
   // `recv_indices` will necessarily be in `indices` on this process, and thus
   // other processes must own them in the submap.
-
+  // If ownership of received index doesn't change, then this process has the
+  // receiving rank as a destination
   std::vector<int> recv_owners(send_disp.back());
+  std::vector<int> submap_dest;
+  submap_dest.reserve(dest.size() + 1);
```

Comment:
Does this `reserve` make sense? How much can we really say about the size of `submap_dest` at this point? It will depend on the indices to include and on how the owners change. Though maybe reserving something in the right ballpark is better for performance...

Reply:
You know that you will have somewhere around the number of destination ranks the original map had. I.e. if you had no dest ranks, you will like not have any new ones now. One reason for reserving is so that the buffer is initialized for MPI communication later on, as modern MPI is not fond of uninitialized buffers.